contents/scalability/scalable-development.md

```diff
@@ -15,12 +15,33 @@ So what is scalability? Scalability is the capacity to which a system can cope w
 
-### Get Better Hardware
+### Horizontal vs Vertical Scaling
+
+#### Horizontal Scaling
+
+Buy more machines to host your server. This is called horizontal scaling. This method can help improve scalability by a huge factor, as each additional unit of processing power and RAM costs the same, no matter how many machines you buy. As this is a promising way to increase scalability, it is no surprise why large services such as Google and Facebook use thousands of computer servers just to provide services to all their users.
+
+However, to fully reap the benefits of multiple machines, you will need to learn various techniques to modify your server infrastructure. In most cases you cannot simply run multiple instances of the server and expect it to work.
+
+Here are some tips to get started with horizontal scaling:
+- Understand parallel computing and how to use it to speed up computation.
+- Learn how to utilize MapReduce on [Hadoop](https://www.tutorialspoint.com/hadoop/index.htm). The Hadoop ecosystem is so versatile and widely-used that there are companies offering Hadoop as a service, so you do not have to set up the server yourself
+- Host your server on [Google App Engine](https://cloud.google.com/appengine/docs). Despite being more expensive than Amazon Web Services and DigitalOcean, it offers very scalable infrastructure for your server.
+- Run a server with multiple nodes with MPI. [Open MPI](https://www.open-mpi.org) is an open source message-passing library that can be used to send data between the nodes in your server. You might have to study [cloud computing infrastructure](http://whatiscloud.com/basic_concepts_and_terminology/it_resource) in order to make use of this.
+- Study various scalable architecture for servers ([resource](http://srinathsview.blogspot.sg/2011/10/list-of-known-scalable-architecture.html)).
+
+#### Vertical Scaling
 
 This is arguably the easiest way to speed up your server, as it is a general solution to most problems. Simply get faster CPUs, larger RAM, etc. This is called vertical scaling. If you get multi-core processors, use threading in your server to allow parallelization of certain processes to take advantage of the cores.
 
 However, due to Amdahl's law, each additional processor will give a less-than-proportionate speed up. Coupled with the exponential increase in price as the computing power of the CPU increases, this is typically not scalable in the long run.
 
+#### Should you choose one or the other?
+
+With vertical scaling, you get to take advantage of processing power and concurrency. But as mentioned above, this isn't sustainable over the long run. Furthermore, you can't run your server on just one machine as it results in a single point of failure and prevents changes to be made to the code without bringing the entire system down. Moreover, you can't distribute your application geographically. 
```

Comment:
This can be done in list form. 

This isn't sustainable over the long run because:
1. Single point of failure
1. Prevents changes made to code...